homeassistant/components/mobile_app/__init__.py

```diff
@@ -69,6 +104,15 @@ async def async_setup(hass: HomeAssistantType, config: ConfigType):
         discovery.async_load_platform(hass, "notify", DOMAIN, {}, config)
     )
 
+    try:
+        conf = config.get(DOMAIN)
+        hass.data[DOMAIN][CONF] = conf
+        if CONF_PUSH in conf:
+            # Create old endpoint for back compatibility with older apps
+            hass.http.register_view(iOSPushConfigView(conf[CONF_PUSH]))
+    except (TypeError, KeyError):
```

Comment:
Just check if key exists.

What can raise type error?

Reply:
> Just check if key exists

I think I did have multiple nested ifs at one point but was looking pretty ugly. See [this diff](https://github.com/home-assistant/core/pull/38333/commits/07bf68507b6024465520004df5fa35ac14bff219#diff-6dcc1a303cf83b8d3bd83e8663097fb7) (which is before I moved from views to webhooks)

> What can raise type error?

The TypeError was covering a NoneType that was returned if `config.get(DOMAIN)` returned a `None` I think. I will see if I can reproduce the cause (Sorry little while back now so need to remember why I had that)

Reply:
Just confirmed if there is no `mobile_app` entry in configuration.yaml we would get a `TypeError` as seen here:

```txt
2020-08-05 12:44:42 ERROR (MainThread) [homeassistant.setup] Error during setup of component mobile_app
Traceback (most recent call last):
  File "/Users/tom/Git/core/homeassistant/setup.py", line 193, in _async_setup_component
    result = await asyncio.wait_for(task, SLOW_SETUP_MAX_WAIT)
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/Users/tom/Git/core/homeassistant/components/mobile_app/__init__.py", line 113, in async_setup
    if CONF_PUSH in conf:
TypeError: argument of type 'NoneType' is not iterable
```

The nested ifs would protect against this but to me seemed to make the code harder to follow. Perhaps a comment in the except explaining where the cases come from?